### PR TITLE
Remove Js Helper from list.

### PR DESCRIPTION
**Pull request by ADmad**
It's a bad practice which we shouldn't promote.

### CODE REVIEW

**Comment by dereuromark** [comment]
I wouldn't call it bad practice to wrap JS into PHP.
It is mainly just overkill, but I also use it (e.g. the GoogleMaps class) to simplify it and just pass arrays (or entities) into it.
So I would not like to remove it just because, if the plugin itself is maintained and working.
I don't know that status, though.

**Comment by dereuromark** [comment]
@oldskool Can you give us your opinion?

**Comment by ADmad** [comment]
Yeah i was just pushing my luck with this :)

**Comment by oldskool** [comment]
I haven't actively maintained my plugin, but that's mainly because there are no reported problems with it :wink: . I'm still ready to fix anything if people report problems or if I discover something myself.

As for the background (if anyone cares :smile:): I primarily "revived" the JsHelper functionality because personally I prefer to use it's `buffer()` method to save up all separate JS calls and render them all at once in a single JS block. I combine this with heredoc syntax highlighting in PhpStorm, so you get "almost" raw JS:

image

I'm not sure why it would be considered bad practice, works like a charm for me without generating any `<script>` block floods/overhead. It also drops the necessity of defining `$(document).ready(function() {});` in every single file, since you all bundle them together in the end. In my layout, I just need one `$this->Js->writeBuffer();` and presto, there it all is, nice and compact in a single script block.

In conclusion:

Personally I see many positive aspects in this approach. This is one of the reasons I love the JsHelper and was sad to see it go in CakePHP 3.0. Therefor I created this plugin and it was linked here (I think by @dereuromark as well?). I'm not sure why it should be (actively) discouraged, but I'd love to hear @AD7six's arguments for that.

**Comment by dereuromark** [comment]
I like the abstraction from JS into PHP in some cases, but that is really super hard to get it working for more than a single use case ;) As we found out with the core helper(s).

You linked the wrong AD* by the way :)
Anyway, let's keep it around for a while, maybe you can add some tests etc. And hopefully someone will pick it up and make it a more useful overall-library then.

**Comment by jadb** [comment]
I am for keeping this. Some people just need basic JS while doing RAD and as long as plugin is maintained, I see no problem advertising it (there are other plugins on the list that aren't really code that should be promoted).

**Comment by ADmad** [comment]
@oldskool Js->writeBuffer() is definitely useful. That's the only Js helper feature is used :)  I agree with @dereuromark and @jadb so closing the PR. 

Please just try to add test cases so that users can use it with confidence. Just need to port the test cases from 2.x :)

**Comment by oldskool** [comment]
@ADmad Good point, I'll put that on my to-do.
